src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/SearchQueryOperation.java

```diff
@@ -21,37 +25,52 @@
 
 @Service
 public class SearchQueryOperation {
+    protected static final String NO_ERROR = null;
     private final MergeDataToSearchResultOperation mergeDataToSearchResultOperation;
     private final GetCaseTypesOperation getCaseTypesOperation;
     private final SearchOperation searchOperation;
+    private final GetDraftsOperation getDraftsOperation;
     private final CaseTypeService caseTypeService;
 
     @Autowired
     public SearchQueryOperation(@Qualifier(CreatorSearchOperation.QUALIFIER) final SearchOperation searchOperation,
                                 final MergeDataToSearchResultOperation mergeDataToSearchResultOperation,
-                                @Qualifier(AuthorisedGetCaseTypesOperation.QUALIFIER) final GetCaseTypesOperation
-                                        getCaseTypesOperation,
+                                @Qualifier(AuthorisedGetCaseTypesOperation.QUALIFIER) final GetCaseTypesOperation getCaseTypesOperation,
+                                @Qualifier(DefaultGetDraftsOperation.QUALIFIER) GetDraftsOperation getDraftsOperation,
                                 final CaseTypeService caseTypeService) {
         this.searchOperation = searchOperation;
         this.mergeDataToSearchResultOperation = mergeDataToSearchResultOperation;
         this.getCaseTypesOperation = getCaseTypesOperation;
+        this.getDraftsOperation = getDraftsOperation;
         this.caseTypeService = caseTypeService;
     }
 
     public SearchResultView execute(final String view,
                                     final MetaData metadata,
                                     final Map<String, String> queryParameters) {
         CaseType validCaseType = caseTypeService.getCaseTypeForJurisdiction(metadata.getCaseTypeId(),
-            metadata.getJurisdiction());
+                                                                            metadata.getJurisdiction());
         Optional<CaseType> caseType = this.getCaseTypesOperation.execute(metadata.getJurisdiction(), CAN_READ)
             .stream()
             .filter(ct -> ct.getId().equalsIgnoreCase(validCaseType.getId()))
             .findFirst();
 
         if (!caseType.isPresent()) {
-            return new SearchResultView(new SearchResultViewColumn[0], new SearchResultViewItem[0]);
+            return new SearchResultView(new SearchResultViewColumn[0], new SearchResultViewItem[0], NO_ERROR);
         }
-        final List<CaseDetails> caseData = searchOperation.execute(metadata, queryParameters);
-        return mergeDataToSearchResultOperation.execute(caseType.get(), caseData, view);
+
+        final List<CaseDetails> cases = searchOperation.execute(metadata, queryParameters);
+
+        String draftResultError = NO_ERROR;
+        List<CaseDetails> draftsAndCases = Lists.newArrayList();
+        try {
+            draftsAndCases = getDraftsOperation.execute(metadata);
```

Comment:
the way it is at the moment draft retrieval and cases retrieval happen sequentially. These 2 searches are independent so can be easily be issued in parallel. This will speed up the overall search. 
what do you think @andrewfolga ?
I appreciate this PR is already big, I can create a separate story for it and I or you can work on that as an improvement later